stdlib/Markdown/src/parse/parse.jl

```diff
@@ -1,5 +1,11 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    MD
+
+`MD` represents a Markdown document. However, `MD` objects should only be constructed using 
```

Comment:
```suggestion
`MD` represents a Markdown document.  Note that the `MD(...)` constructor should not generally be used
directly, since it constructs the internal data structures.   Instead, you can construct `MD` objects using
```
Good here to be explicit that we are warning against calling the `MD(...)` constructor directly.

For example, there are also the `Markdown.parse` functions that we might want to document at some point.

Alternatively, we might want to define:
```jl
MD(md::AbstractString) = parse(md)
```
since as far as I can tell there is no need to ever call the constructor with a single string argument — you generally want to to have at least a `Markdown.Paragraph` object, and in any case you can always call the lower-level constructor `MD([str])` in the event that you really do want the `content` to consist of a single string.  However, this is a change that should go in a separate PR.

Reply:
I agree with documenting the `Markdown.parse` function, although I'm guessing that this should then include a description of the entire parse logic, including the `Markdown.@flavor` macro. I'm just not sure whether this should be part of the public API or part of the developer documentation.

Reply:
If we are providing a Markdown module, to me it's pretty basic functionality to document how to parse strings generated at runtime (as opposed to literal strings).  This isn't a low-level thing, like the internals of the `MD` type.

That being said, I agree that the flavors open a can of worms, so it might be better to leave this for a separate issue/PR so that this one can be merged.